packages/viewport/src/test/if-viewport-matches.js

```diff
@@ -1,58 +1,45 @@
 /**
  * External dependencies
  */
-import TestRenderer, { act } from 'react-test-renderer';
+import { render, screen } from '@testing-library/react';
 
 /**
  * WordPress dependencies
  */
-import { useViewportMatch as useViewportMatchMock } from '@wordpress/compose';
+import { useViewportMatch } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import '../store';
 import ifViewportMatches from '../if-viewport-matches';
 
+jest.mock( '@wordpress/compose/src/hooks/use-viewport-match' );
+
 describe( 'ifViewportMatches()', () => {
 	const Component = () => <div>Hello</div>;
 
 	afterEach( () => {
-		useViewportMatchMock.mockClear();
+		useViewportMatch.mockClear();
 	} );
 
 	it( 'should not render if query does not match', () => {
-		useViewportMatchMock.mockReturnValueOnce( false );
+		useViewportMatch.mockReturnValueOnce( false );
 		const EnhancedComponent = ifViewportMatches( '< wide' )( Component );
+		render( <EnhancedComponent /> );
 
-		let testRenderer;
-		act( () => {
-			testRenderer = TestRenderer.create( <EnhancedComponent /> );
-		} );
-
-		expect( useViewportMatchMock.mock.calls ).toEqual( [
-			[ 'wide', '<' ],
-		] );
+		expect( useViewportMatch ).toHaveBeenCalledWith( 'wide', '<' );
 
-		expect( testRenderer.root.findAllByType( Component ) ).toHaveLength(
-			0
-		);
+		expect( screen.queryByText( 'Hello' ) ).not.toBeInTheDocument();
 	} );
 
 	it( 'should render if query does match', () => {
-		useViewportMatchMock.mockReturnValueOnce( true );
+		useViewportMatch.mockReturnValueOnce( true );
 		const EnhancedComponent = ifViewportMatches( '>= wide' )( Component );
-		let testRenderer;
-		act( () => {
-			testRenderer = TestRenderer.create( <EnhancedComponent /> );
-		} );
-
-		expect( useViewportMatchMock.mock.calls ).toEqual( [
-			[ 'wide', '>=' ],
-		] );
-
-		expect( testRenderer.root.findAllByType( Component ) ).toHaveLength(
-			1
-		);
+		render( <EnhancedComponent /> );
+
+		expect( useViewportMatch ).toHaveBeenCalledWith( 'wide', '>=' );
+
+		expect( screen.queryByText( 'Hello' ) ).toBeInTheDocument();
```

Comment:
Just wanted to point out that `query*` queries are recommended only when we're querying for elements that don't exist. In case they are expected to exist, we use `get*` queries. See [this post](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-query-variants-for-anything-except-checking-for-non-existence) from the library creator as the rationale for that.

Reply:
Thanks, @tyxla.

I just realized that after reading https://testing-library.com/docs/react-testing-library/cheatsheet#queries.